modules/public/views.py

```diff
@@ -162,28 +162,70 @@ def convert_limit_offset_to_size_page(link):
 
 
 def homepage():
+    featured_snaps = []
+    errors = []
+    status_code = 200
+    try:
+        featured_snaps = normalize_searched_snaps(api.get_featured_snaps())
+    except api.ApiResponseError as api_error_exception:
+        if hasattr(api_error_exception, 'errors'):
+            errors = api_error_exception.errors
```

Comment:
You simply want to pass all errors from the API straight onto your `errors` object? I'm not sure that's a good idea. There are a couple of ways in which you could get multiple errors, I believe:

- You may be making multiple API calls per view function
- You may be getting multiple errors back from a single API call

I don't think you should merge the two sets of errors. I think it makes more sense to preserve that hierarchy. Here's two ways I can think of to do that:

``` python
error_messages = []

try:
    api.get_one()
except ApiError as api_error:
    error_messages.append("API request 'one' gave us an error. Naughty API.")

try:
    api.get_two()
except ApiResponseErrorList as list_error:
    error_codes = ", ".join(list_error.errors.keys())
    error_messages.append("API request 'two' gave us a bunch of errors: " + error_codes)
```

or

``` python
errors = []

try:
    api.get_one()
except ApiError as api_error:
    errors.append({'message': "API request 'one' gave us an error. Naughty API."})

try:
    api.get_two()
except ApiResponseErrorList as list_error:
    errors.append({'message': "API request 'two' gave us a bunch of errors", 'errors': list_error.errors)
```